### DISCUSSION TITLE
Reposition Transform Buttons Upon Selection Based On Selection Location.

### DISCUSSION THREAD

**Discussion by amaxyza** [ppy/osu#23599] [Ideas] [answered]
Upon selection, certain transformation options are provided directly underneath the selection box. However, they are always underneath the selection, even if the selection happens to be near the bottom of the playfield. This will cause the transformation buttons to be unusable by the user until the selection is moved upward. 

Here is an image showcasing what I am referring to:

![image](https://github.com/ppy/osu/assets/116742308/9b25de3f-41f7-410a-a1c6-dae027147248)

I have looked for quite some time to add a suggestion for code myself, but I cannot seem to find where the code for drawing the button graphics are, nor do I possess the codebase knowledge to make such changes. My suggestion is after a certain threshold of distance away from the center of the screen the origin of the selection lies on, the transform buttons flip from the bottom of the selection to the top, for convenience. This would completely remove the step of having to reposition the selection in order to use the functionality provided by the buttons, which currently do not have shortcut alternatives (to my knowledge). 

Here is a very scuffed example of what I would like to see in the osu!lazer editor per my suggestion; 

![image](https://github.com/ppy/osu/assets/116742308/56e2fd23-830c-499c-bf59-ef9856099b84)

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/pull/23649

**Comment by cihe13375** [comment]
This still breaks in some cases. Tested on 2023.621.0.
![case1](https://github.com/ppy/osu/assets/33959010/329c83a1-0e3e-4b91-8079-72887811754b)
![case2](https://github.com/ppy/osu/assets/33959010/82086d06-9456-43b0-ad04-b9889a1c9fa8)

**Reply by peppy** [reply]
Could you please provide un-cropped screenshots?

**Reply by cihe13375** [reply]
![1111](https://github.com/ppy/osu/assets/33959010/9fb31a42-5831-44ae-98c0-6eeb658e6e7f)
![2222](https://github.com/ppy/osu/assets/33959010/d55f7372-ae04-4df8-9c39-d88f879e8e0f)

**Reply by peppy** [reply]
@bdach gonna leave this one with you.

**Reply by bdach** [reply]
Addressed via https://github.com/ppy/osu/pull/24099
